### PR TITLE
Refactor CI scripts

### PR DESCRIPTION
**Pull request by BenjaminW3**
* install and execute shellcheck and sloc also on mac
* Split generate and build phase as this is more idiomatic

### CODE REVIEW

File: script/install_analysis.sh [hunk 1]
```diff
@@ -14,12 +14,28 @@ source ./script/travis_retry.sh
...
+    # Install sloc
+    brew install sloccount
+    sloccount --version
+
+    #-------------------------------------------------------------------------------
+    # Install shellcheck
+    brew install shellcheck
+    shellcheck --version
+
+fi
```
Comment (psychocoderHPC) [inline script/install_analysis.sh]: Could you please at a empty line.

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]
only a little change needed

File: script/install_analysis.sh (hunk 1 above)
Comment (BenjaminW3) [reply script/install_analysis.sh]: done

**Comment by BenjaminW3** [comment]
Please approve again. Build is green now.

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**